frontend/src/pages/profile/ProfileEditPopup.tsx

```diff
@@ -0,0 +1,277 @@
+import {
+  Dialog,
+  DialogHeader,
+  DialogBody,
+  DialogFooter,
+} from '@material-tailwind/react';
+import type { Personnel } from '../dashboard';
+import { ButtonTypes } from '@/common';
+import { EditProfileValidationSchema, fields, sections } from './constants';
+import { Divider } from '@/components/ui/Divider';
+import type { FormikHelpers, FormikProps, FormikState } from 'formik';
+import { Field, Form, Formik } from 'formik';
+import { Button, SectionHeader, Select, TextInput } from '@/components';
+import { AxiosPrivate } from '@/utils';
+import { format } from 'date-fns';
+
+export const ProfileEditPopup = ({
+  open,
+  handleOpenEditPopUp,
+  personnel,
+  updatePersonnel,
+}: {
+  open: boolean;
+  handleOpenEditPopUp: () => void;
+  personnel: Personnel;
+  updatePersonnel: (personnel: Personnel) => void;
+}) => {
+  const initialValues: Personnel = { ...personnel };
+
+  delete initialValues?.availability;
+  delete initialValues?.experiences;
+  const stringToBoolean = (value: string): boolean => {
+    switch (value) {
+      case 'true':
+        return true;
+      case 'false':
+        return false;
+      default:
+        return false;
+    }
+  };
+  const handleSubmit = async (
+    values: Personnel,
+    helpers: FormikHelpers<Personnel>,
+  ) => {
+    // only send the fields that have been changed
+    values.remoteOnly = stringToBoolean(values.remoteOnly ? 'true' : 'false');
```

Comment:
These string to booleans feel like they should be unnecessary.. Why are they necessary?

Reply:
I'm using a Field from formik, the type for a value is "string|boolean|number". Even though the value is a boolean, when it's passed into the submit function it is a string, so I'm converting it to a boolean.